src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,87 @@
+package duke;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+
+public class TaskList {
+    private List<Task> tasks;
+    private Storage storage;
+
+    /**
+     * Constructor for TaskList class.
+     * @param storage
+     * @throws FileNotFoundException
+     * @throws IOException
+     * @throws DukeException
+     */
+    public TaskList(Storage storage) throws DukeException {
+        this.storage = storage;
+        this.tasks = this.storage.getTasksFromFile();
+    }
+
+    /**
+     * Append Task to the list of tasks, and saves file to storage.
+     * @param task Task to be added.
+     * @throws DukeException If there is a generic Duke Exception.
+     */
+    public void add(Task task) throws DukeException {
+        this.tasks.add(task);
+        this.storage.saveToFile(tasks);
+    }
+
+    public Task get(int idx) {
+        return this.tasks.get(idx);
+    }
+
+    public int size() {
```

Comment:
Perhaps it might be better to start the method name with a verb, such as getSize()